### PR TITLE
check dim and order in inner constructor

### LINKED ISSUE #63: rand(Tensor{3,3}) should fail
```jl
julia> rand(Tensor{3,3});

julia> rand(Tensor{3,3})
Error showing value of type ContMechTensors.Tensor{3,3,Float64,27}:
.
.
```

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes #63 

Need to check performance but I will do that on work station. The check should likely be constant folded.

### CODE REVIEW

File: src/ContMechTensors.jl [hunk 1]
```diff
@@ -17,6 +15,10 @@ export setindex, tdot, dotdot
 
 @deprecate extract_components(tensor) Array(tensor)
 
+const SYMMETRIC_ORDER_ERROR_STRING = "only tensors of order 2, 4 supported for symmetric tensors"
```
Comment (fredrikekre) [inline src/ContMechTensors.jl:18]: ... order 2 and 4 maybe? same for the two lines below.

**Review by fredrikekre — APPROVED** [review]

**Comment by KristofferC** [comment]
https://gist.github.com/KristofferC/3bd9407af2fd934e82c7820a8f36e828

:fearful:

**Comment by fredrikekre** [comment]
This is just from the dimension checking? :/

**Comment by KristofferC** [comment]
Yes, it prevents some inlining. Even with an explicit `@inline` it still does bad stuff. Hmm.

**Comment by fredrikekre** [comment]
We could add this for all different cases? :smile: 

``` jl
(::Type{SymmetricTensor{2, 1, T, 1}}){T <: Real, T2}(t::NTuple{1, T2}) = SymmetricTensor{2, 1, T, 1}(SVector{1,T}(t))

(::Type{SymmetricTensor{2, 2, T, 3}}){T <: Real, T2}(t::NTuple{3, T2}) = SymmetricTensor{2, 3, T, 1}(SVector{3,T}(t))
...
```

**Comment by KristofferC** [comment]
It doesn't help when the tensor is created from an `SVector` directly.

Easy way would be that all the ways of constructing a tensor does the check, but it is a bit annoying.

**Comment by fredrikekre** [comment]
Do we care about that?

**Comment by KristofferC** [comment]
It is an easy mistake if you mix up the order and the dimension.

**Comment by fredrikekre** [comment]
But if we don't define `Tensor` constructors that allow for such mistakes?

**Comment by fredrikekre** [comment]
Nevermind

**Comment by KristofferC** [comment]
Done properly in #65

**Pull request closed without merging**